crates/biome_js_analyze/src/lint/nursery/use_at_index.rs

```diff
@@ -0,0 +1,927 @@
+        make::js_number_literal_expression(make::js_number_literal(value)),
+    ))
+}
+
+/// check if the node is a slice
+/// # Examples
+/// ```js
+///     .slice(0)[0]
+///     .slice(0, 1).pop(0)
+/// ```
+fn analyze_slice_element_access(node: &AnyJsExpression) -> Option<UseAtIndexState> {
+    if is_within_delete_expression(node).unwrap_or(false) {
+        return None;
+    }
+    // selector
+    let (selected_exp, extract_type): (AnyJsExpression, SliceExtractType) = match node {
+        // .pop() or .shift()
+        AnyJsExpression::JsCallExpression(call_exp) => {
+            let arg_length = call_exp.arguments().ok()?.args().into_iter().count();
+            if arg_length != 0 {
+                return None;
+            }
+            let member = solve_parenthesized_expression(call_exp.callee().ok()?)?;
+            let AnyJsExpression::JsStaticMemberExpression(member) = member else {
+                return None;
+            };
+            if call_exp.is_optional_chain() || member.is_optional_chain() {
+                return None;
+            }
+            let member_name = member
+                .member()
+                .ok()?
+                .as_js_name()?
+                .value_token()
+                .ok()?
+                .token_text_trimmed();
+            let object = solve_parenthesized_expression(member.object().ok()?)?;
+            if member_name == "pop" {
+                (object, SliceExtractType::Pop)
+            } else if member_name == "shift" {
+                (object, SliceExtractType::Shift)
+            } else {
+                return None;
+            }
+        }
+        AnyJsExpression::JsComputedMemberExpression(member) => {
+            let object = solve_parenthesized_expression(member.object().ok()?)?;
+            if member.is_optional_chain() {
+                return None;
+            }
+            let value =
+                get_integer_from_literal(&solve_parenthesized_expression(member.member().ok()?)?)?;
+            // enable only x[0]
+            if value != 0 {
+                return None;
+            }
+            (object, SliceExtractType::ZeroMember)
+        }
+        _ => return None,
+    };
+    // .slice(0,1)
+    let AnyJsExpression::JsCallExpression(call_exp) = selected_exp else {
+        return None;
+    };
+    let AnyJsExpression::JsStaticMemberExpression(member) = call_exp.callee().ok()? else {
+        return None;
+    };
+    let member_name = member
+        .member()
+        .ok()?
+        .as_js_name()?
+        .value_token()
+        .ok()?
+        .token_text_trimmed();
+    if member_name != "slice" {
+        return None;
+    }
+    // arg length should be 1 or 2
+    let args: Vec<_> = call_exp
+        .arguments()
+        .ok()?
+        .args()
+        .into_iter()
+        .flatten()
+        .collect();
+    if args.is_empty() || args.len() > 2 {
+        return None;
+    }
```

Comment:
We have a dedicated function that should simplify retrieving arguments:
```suggestion
    let [Some(arg0), optional_arg1, None] = call_exp
        .arguments()
        .ok()?
        .get_arguments_by_index([0, 1, 2])) else {
        return None;
    };
```

This also avoid the allocation of a `Vec`